browser/ui/brave_shields_data_controller.cc

```diff
@@ -42,6 +56,23 @@ void BraveShieldsDataController::DidFinishNavigation(
 void BraveShieldsDataController::WebContentsDestroyed() {
   favicon::ContentFaviconDriver::FromWebContents(web_contents())
       ->RemoveObserver(this);
+  observation_.Reset();
+}
+
+void BraveShieldsDataController::OnContentSettingChanged(
+    const ContentSettingsPattern& primary_pattern,
+    const ContentSettingsPattern& secondary_pattern,
+    ContentSettingsTypeSet content_type_set) {
+  // This callback is only used for notifying changes from different tab of
```

Comment:
why would we have a different mechanism for this tab vs other tabs? That seems to unnecessarily complicate things